src/lib/components/table/table-settings/table-settings.component.spec.ts

```diff
@@ -0,0 +1,90 @@
+import {
+    Directive,
+    Input
+} from '@angular/core';
+import {
+    ComponentFixture,
+    TestBed
+} from '@angular/core/testing';
+import { TranslationModule } from 'angular-l10n';
+import {
+    MatDialog,
+    MatDialogRef
+} from '@angular/material/dialog';
+import { MatButtonModule } from '@angular/material/button';
+import { TableSettingsComponent } from './table-settings.component';
+import { of } from 'rxjs';
+
+@Directive({selector: '[tcTooltip]'})
+class MockTooltipDirective
+{
+    @Input('tcTooltip')
+    public tooltip:string;
+}
+
+describe('TableSettingsComponent', () =>
+{
+    let component:TableSettingsComponent;
+    let fixture:ComponentFixture<TableSettingsComponent>;
+
+    let mockDialogRef:Partial<MatDialogRef<any>> = {
+        afterClosed: () => of(['four', 'five'])
+    };
+    let mockDialog:Partial<MatDialog> = {
+        open: () => mockDialogRef as MatDialogRef<any>,
+    };
+
+    beforeEach(() =>
+    {
+        TestBed.configureTestingModule(
+            {
+                declarations: [
+                    TableSettingsComponent,
+                    MockTooltipDirective
+                ],
+                imports:      [
+                    TranslationModule.forRoot({}),
+                    MatButtonModule
+                ],
+                providers:    [
+                    {
+                        provide:  MatDialog,
+                        useValue: mockDialog
+                    },
+                    {
+                        provide: MatDialogRef,
+                        useValue: mockDialogRef
+                    }
```

Comment:
Das brauchst du nicht

Reply:
NullInjectorError: StaticInjectorError[MatDialogClose -> MatDialogRef]: 
          NullInjectorError: No provider for MatDialogRef!

Reply:
nevermind - fixed it